fixed issues with plan create page

briri opened the pull request:
addresses plan creation issues on #777 
- separated out shared controller endpoint that was handling ajax template retrieval and form submission
- added error handler to display notice to user when no templates were found
- disable submit button until a template has been returned
- adjusted integration tests to work with new json endpoint

jollopre commented on app/controllers/plans_controller.rb:
```diff
@@ -330,6 +332,52 @@ def set_test
 
+  # Collect all of the templates available for the org+funder combination
+  # --------------------------------------------------------------------------
+  def template_options()
```
Have you considered moving this logic to the templates model?

briri replied:
yes, that would be more appropriate

jollopre commented on lib/assets/javascripts/views/plans/new.js:
```diff
@@ -49,14 +62,14 @@ $(() => {
       $.ajax({
-        method: getMethod(jQueryForm),
-        url: getAction(jQueryForm),
+        method: 'POST',
```
why are you hardcoding the method, could not you get it from the form method?

briri replied:
Its not using the form's `action` in this case so I was reluctant to assume that the method would be the same.

jollopre commented on lib/assets/javascripts/views/plans/new.js:
```diff
@@ -1,17 +1,32 @@
   // AJAX error function for available template search
   const error = () => {
-    // TODO: add error handling
+    renderAlert(NO_TEMPLATE_FOUND_ERROR);
```
Nice to see here the usage of notification  messages through JS too

briri commented:
- moved template_options endpoint from the plans controller to template controller
- switched to the more appropriate GET method instead of POST
- also rebased with latest from CDL-MVP

jollopre commented:
Please, fix the merge conflicts and merge back yourself
